src/lib/proof-system/zkprogram.ts

```diff
@@ -110,18 +110,21 @@ class Proof<Input, Output> {
       proof: Pickles.proofToBase64([this.maxProofsVerified, this.proof]),
     };
   }
-  static fromJSON<S extends Subclass<typeof Proof>>(
+  static async fromJSON<S extends Subclass<typeof Proof>>(
```

Comment:
`Proof.fromJSON()` becomes async because it needs `Pickles` and there's the chance people call this without calling `compile()` first